pyclowder/connectors.py

```diff
@@ -519,8 +519,12 @@ def register_extractor(self, endpoints):
 
         for url in endpoints.split(','):
             if "unique_key" in data:
-                if url.find("?") > -1: url += "&user=%s" % self.clowder_email
-                else: url += "?user=%s" % self.clowder_email # TODO: This will not work, need an auth key matching email
+                if url.find("?") > -1:
```

Comment:
what if this is the first parameter?

Reply:
if the config has specified a registration URL without an API key, registering privately with just an email address won't be accepted. so it doesn't send the request.